src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            tail = new Node<>(null, value, null);
+            head = tail;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index >= 0 && index < size) {
+            linkMiddle(value, getNodeByIndex(index));
+        } else {
+            checkElementIndex(index);
+        }
     }
```

Comment:
I think you can remove "else if" block and in "else" block after you've checked index you can use linkMiddle(value, getNodeByIndex(index));

Reply:
great advice, it works